be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -99,16 +99,26 @@ Status HdfsScannerCSVReader::_fill_buffer() {
 Status HdfsTextScanner::do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     TTextFileDesc text_file_desc = _scanner_params.scan_ranges[0]->text_file_desc;
     _field_delimiter = text_file_desc.field_delim;
-    // we should cast string to char now since csv reader only support record delimiter by char
+
+    // we should cast string to char now since csv reader only support record, collection, mapkey delimiter by char
     _record_delimiter = text_file_desc.line_delim.front();
+    _collection_delimiter = text_file_desc.collection_delim.front();
+    _mapkey_delimiter = text_file_desc.mapkey_delim.front();
```

Comment:
Should check if the string size is 1 before calling `std::string::front()` otherwise you will get unexpected delimiters.

Reply:
I had checked it in `HdfsTextScanner::do_init()` method.

Reply:
Could you please give me a reference link?  On the other hand, do not rely on implicit context, either add explicit if checks or add assertions here.

Reply:
Maybe I need to move `DCHECK()` before here to ensure all `delim` will not be empty.

The reason to use `std::string` here is that all `delim` is passed through Thrift, and all fields in Thrift are string.

Another reason for `_collection_delimiter` and `_mapkey_delimiter` use string is to keep the same behavior with existed `_field_delimiter` and `_record_delimiter`.

Reply:
You have to ensure the string length is exactly one, not just not empty.

Reply:
IMO, you should define `_field_delimiter` and `_record_delimiter` of type `char` for consistent style and not the other way around. 
People who read these codes is easy to feel confused why can you extract the first character of string as delimiter without checking the string length first.

Reply:
I fix it.